Yuidocjs integration

FabianFrank opened the pull request:
Migrated from Python yuidoc to YUIDoc.js, also added a --server switch for the new server feature of YUIDoc.js. Also removed all async shell calls from the doc generation process and replaced them with sync syscalls.

Utilized https://github.com/ryanmcgrath/wrench-js to avoid reinventing the wheel when re-implementing recursive fs operations.

This pull request resolves issue #237 and issue #84.

FabianFrank committed:
Porting mojito doc over to the new yuidocjs

Ported mojito doc to use the new yuidocjs from https://github.com/yui/yuidoc. Replacing unnecessary shell calls with calls to JavaScript syscall wrappers. Utilizing wrench for recursive fs operations: https://github.com/ryanmcgrath/wrench-js. Deleting old python yuidoc from the repo.

FabianFrank committed:
Adding YUIDoc server support

Add --server option to mojito docs. This allows to run a server locally that serves the documentation instead of writing it out to disk. This works for apps, mojits as well as the mojito api doc itself.

drewfish commented on source/package.json:
```diff
@@ -20,7 +20,9 @@
         "yui": "~3.5.1",
-        "yuitest": "~0.7.4"
+        "yuitest": "~0.7.4",
+        "wrench": "~1.3.9",
```
I wonder if "wrench" and "yuidocjs" should be in "devDependencies" instead?  (That is, neither of those are needed in production.)

Hmmm... perhaps there should be a separate "mojito-dev" (or "mojito-tools") package that contains development-time things (and let "mojito" just contain the runtime parts).

drewfish replied:
Oh yeah, for sanity please keep the dependencies sorted.

FabianFrank replied:
Those are good points. I did not realize that the dependencies were sorted, sorry. I agree with your assessment and I will move them to devDependencies.

FabianFrank committed:
cleaning up dependencies

wrench and yuidoc are only needed for development, but not to run the app in production. so moving them to devDependencies.

FabianFrank committed:
Merge branch 'master' of github.com:yahoo/mojito into yuidocjs

isao commented:
secondary failed (pull job 30) because wrench isn't whitelisted internally

even so :+1: :+1:

FabianFrank commented:
I requested wrench to be whitelisted internally.

FabianFrank commented:
@isao wrench has been whitelisted internally, so the secondary build should also pass now. can you please check again? thanks!

isao commented:
nice. mridgeway's processing PRs this sprint, assigning..

Fabian, can you give him a heads up on any license/manifest/readme changes we need as well?

FabianFrank commented:
I don't think we have to make changes to the README or other documentation, because the usage of mojito docs didn't change. wrench is MIT licensed, like all other npm modules that we use. We could mention it at the bottom of the README, like we mention Mustache. On the other hand we also don't mention other dependencies there, so I wasn't sure if we want this. Any thoughts?

FabianFrank commented:
@isao @mridgway Is there an ETA for when we can merge this one?

mridgway commented:
@FabianFrank I'll get this fixed up and in by the end of the week.

mridgway commented:
Merged. Thanks a bunch @FabianFrank.
